common/lifecyleMgr.go

```diff
@@ -108,24 +113,62 @@ func (lcm *lifecycleMgr) watchInputs() {
 			continue
 		}
 
-		// reads input until the first occurrence of \n in the input,
-		input, err := consoleReader.ReadString('\n')
-		timeReceived := time.Now()
+		//if we allow cancelFromStdin, the json messages will not be allowed
```

Comment:
why not? StgExp will use both most likely

Reply:
We do not have an easy way to peek stdin to check if the message is cancel or a new json struct type. json decoder will not work with partial strings, and we would give up ease of using the library if we parse it ourselves. Given stegExp will be upgrading tto this feature, it would be reasonable to ask them to move to new format of input on cancel as well.